src/client/components/SearchAPI/Fetcher.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable react/prop-types */
+import React, { useState, useEffect } from 'react';
+
+const baseUrl = 'https://api.github.com/search/users?q=';
+
+export const Fetcher = ({ search, render }) => {
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState(search);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const getData = async () => {
+      setLoading(true);
+      const response = await fetch(`${baseUrl}${encodeURIComponent(search)}`);
```

Comment:
To make this component reusable, you can move the `fetch` code to a parameter - this way you can mock in storybook using a fake function for example.